examples_tests/41.VisibilityBuffer/main.cpp

```diff
@@ -98,6 +98,100 @@ vec3 nbl_glsl_fetchVtxNormal(in uint vtxID)
 
 )";
 
+constexpr const char* FRAGMENT_SHADER_OVERRIDES = //also turns off set3 bindings (textures) because they're not needed anymore as we're using VT
+R"(
+#ifndef _NO_UV
+    #include <nbl/builtin/glsl/virtual_texturing/extensions.glsl>
+
+    #define _NBL_VT_DESCRIPTOR_SET 0
+    #define _NBL_VT_PAGE_TABLE_BINDING 0
+
+    #define _NBL_VT_FLOAT_VIEWS_BINDING 1 
+    #define _NBL_VT_FLOAT_VIEWS_COUNT %u
+    #define _NBL_VT_FLOAT_VIEWS
+
+    #define _NBL_VT_INT_VIEWS_BINDING 2
+    #define _NBL_VT_INT_VIEWS_COUNT 0
+    #define _NBL_VT_INT_VIEWS
+
+    #define _NBL_VT_UINT_VIEWS_BINDING 3
+    #define _NBL_VT_UINT_VIEWS_COUNT 0
+    #define _NBL_VT_UINT_VIEWS
+    #include <nbl/builtin/glsl/virtual_texturing/descriptors.glsl>
+
+    layout (set = 2, binding = 0, std430) restrict readonly buffer PrecomputedStuffSSBO
+    {
+        uint pgtab_sz_log2;
+        float vtex_sz_rcp;
+        float phys_pg_tex_sz_rcp[_NBL_VT_MAX_PAGE_TABLE_LAYERS];
+        uint layer_to_sampler_ix[_NBL_VT_MAX_PAGE_TABLE_LAYERS];
+    } precomputed;
+#endif
+#define _NBL_FRAG_SET3_BINDINGS_DEFINED_
+
+struct PCstruct
+{
+    vec3 Ka;
+    vec3 Kd;
+    vec3 Ks;
+    vec3 Ke;
+    uvec2 map_Ka_data;
+    uvec2 map_Kd_data;
+    uvec2 map_Ks_data;
+    uvec2 map_Ns_data;
+    uvec2 map_d_data;
+    uvec2 map_bump_data;
+    float Ns;
+    float d;
+    float Ni;
+    uint extra; //flags copied from MTL metadata
+};
+layout (push_constant) uniform Block {
+    PCstruct params;
+} PC;
+#define _NBL_FRAG_PUSH_CONSTANTS_DEFINED_
```

Comment:
remember to pack it into an SSBO instead, but remember to leave `#define _NBL_FRAG_PUSH_CONSTANTS_DEFINED_` so there are no definitions of unused push constants

Reply:
I need to use push constants anyway, see last commit